pkg/api/validate.go

```diff
@@ -10,8 +10,8 @@ import (
 )
 
 var (
-	rxRfc1123 = regexp.MustCompile(`(?i)^` +
-		`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+	rxRfc952 = regexp.MustCompile(`(?i)^` +
+		`([a-z][-a-z0-9]{0,61}[a-z0-9])` +
 		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
 		`$`)
 
```

Comment:
Are we saying A-Z is not allowed in all blocks? `labels may contain only the ASCII letters 'a' through 'z' (in a case-insensitive manner), the digits '0' through '9', and the minus sign ('-')`
Is the last part `{0.61}` is right? I understand this is `.com` part. There should be the limit for this too? 

This is really hard to understand I would like to see some simple unit test for this. And some comments on the "what's behind the regexp".

Reply:
Added testing.

Reply:
I'm not sure this is actually RFC 952 compliant anyway.  952 specifies that the string can only be up to 24 characters.  Kube used to validate this though I think they may have changed.  Here is an old commit with the expression they were using: https://github.com/kubernetes/kubernetes/commit/54a3efb36428b6d77081d9a997b6b233f0e710e4#diff-4cb64063f15b780708c65e9c71922a5dR124 

https://tools.ietf.org/html/rfc952

Reply:
Original Error:
```
 Service returned an error. Status=400 Code="InvalidDomainNameLabel" Message="The domain name label 7e37591a6e974a0f93af is invalid. It must conform to the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$
```
It appears that the regex is just checking the first part of the string.  The storage account is the `7e37...` string.  So it might not be completely rfc952 compliant.  Should we update the regular expression variable name to be `rxHostname` instead of `rfc952`?

Reply:
Urg, standards.

OK, so we are going to need a sane isValidHostname for RouterProfile.PublicSubdomain (and in the future, Properties.PublicHostname).  So let's just leave rxRfc1123 and isValidHostname right where they are.

Now, for properties.FQDN and routerProfile.FQDN.  These should validate against Azure's reading of the spec -- let's just use the regex they kindly gave us.

Thus: in isValidCloudAppHostname, stop calling isValidHostname but instead check that the first part matches Azure's regexp `(?i)^[a-z][-a-z0-9]{1,61}[a-z0-9]$`.  Continue to the suffix as we do now.